src/oscar/apps/dashboard/ranges/views.py

```diff
@@ -143,28 +162,68 @@ def add_products(self, request):
             return self.render_to_response(ctx)
 
         self.handle_query_products(request, range, form)
-        self.handle_file_products(request, range, form)
-        return HttpResponseRedirect(self.get_success_url(request))
+        self.handle_file_products(request, range, form, included=True)
+        return redirect(reverse('dashboard:range-products',
+                                kwargs={'pk': range.pk}))
+
+    def add_excluded_products(self, request):
+        range = self.get_range()
+        form = self.form_class(
```

Comment:
Here you can remove the `excluded=True` as you have access to the upload type:
form.cleaned_data["upload_type"] and thus you also know what the active tab will be.